qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -397,11 +403,16 @@ def pass_manager(self, pass_manager_config, optimization_level=None) -> PassMana
             pass_manager_config.initial_layout,
         )
         if optimization_level == 0:
+            if CONFIG.get("sabre_all_threads", None) or os.getenv("QISKIT_SABRE_ALL_THREADS"):
+                trial_count = CPU_COUNT
+            else:
+                trial_count = 5
+
```

Comment:
It would be nice to replace this repeating block with a small util function like this:

```python
def get_trial_count(default_trials=5):
    if CONFIG.get("sabre_all_threads", None) or os.getenv("QISKIT_SABRE_ALL_THREADS"):
        return CPU_COUNT
    return default_trials
```

And use it here and in the other places below

Reply:
This is a good idea, I'll refactor this. I was also thinking about this logic overnight and was wondering if maybe it would make more sense to `max(default_trials, CPU_COUNT)` if the option is set instead of always `CPU_COUNT` . Then we don't need to caveat about not setting the flag if you have less than 20 CPUs on your local system.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/12780/commits/21ff70fc5f7c09521a3ca90a117030d2b7b7349e